pkg/machinery/node.go

```diff
@@ -29,3 +29,12 @@ func (m *machinery) IsNodeCordoned(node *corev1.Node) *IsCordonedResult {
 		AddedAt:    cordonAddedTime,
 	}
 }
+
+// IsNodeUpgrading returns bool
+func (m *machinery) IsNodeUpgrading(node *corev1.Node) bool {
+	if node.Annotations["machineconfiguration.openshift.io/currentConfig"] != node.Annotations["machineconfiguration.openshift.io/desiredConfig"] {
```

Comment:
I think it's better to check if the `machineconfiguration.openshift.io/state` annotation value is `Working` which AFAIK implies that the configs are validated and there are no issues found at MCD level which otherwise would cause the state to be `Degraded`. 

What I mean is, when the current and desired config is not the same, the state annotation could either be `Working` or `Degraded`. But if for any reason MCO/MCD is not able to reboot the node at all since node is Degraded, at this point SRE intervention is required before it is qualified for upgrade. 

If it helps, we can run a quick test with two scenarios:

1.  `oc debug` into a node and intentionally change a file which is managed by MCO and let the node become `Degraded`. At this time, upgrade should get stuck or SRE should get some alert when the current and desired config is not the same but the node update is stuck too.
2. Second test is the usual one where node is all fine and the usual upgrade at MCO level is marked by state being `Working`. 

(When I say state, I mean `machineconfiguration.openshift.io/state` annotation)

Let me know your thoughts on same.

Reply:
Thanks, Ravi for the review. I updated the requested change in my latest commit